pkg/agent/proxy/proxier.go

```diff
@@ -482,6 +485,13 @@ func (p *proxier) installServices() {
 							continue
 						}
 					}
+					// If previous Service which has ClusterIP should be removed, remove ClusterIP routes.
+					if svcInfo.ClusterIP() != nil && utilruntime.IsWindowsPlatform() {
+						if err := p.deleteClusterIPServiceRoutes(pSvcInfo.ClusterIP()); err != nil {
+							klog.ErrorS(err, "Failed to remove ClusterIP Service routes", "Service", svcPortName)
+							continue
+						}
+					}
```

Comment:
In theory there should be no difference between Windows and Linux implementation here. But it seems it has difference because you are using different ways to install service routes. We can unify later but I would suggest not to make them different in caller. You can always call `p.routeClient.DeleteClusterIPRoute` and have a dummy implementation for Linux.
And I don't think `p.deleteClusterIPServiceRoutes` is more readable than `p.routeClient.DeleteClusterIPRoute`.

Reply:
Used `p.routeClient.DeleteClusterIPRoute` now. Please check again if it is what you expect.

Reply:
Please check the first paragraph of the comment as well.

Reply:
Maybe there's some misunderstanding. I am using same logic to install Service routes. The difference is that Linux doesn't seem to delete such routes, which I don't know why but Windows will do.